.github/workflows/ci.yml

```diff
@@ -75,3 +78,35 @@ jobs:
       - name: Run CI (macOS)
         if: matrix.os == 'macos-latest'
         run: make ci.macos
+  staticcheck:
+    name: 'Run staticcheck'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ 'ubuntu-latest', 'macOS-latest' ]
+        go: [ '1.17', '1.18', '1.19', '1.20' ]
+        dir: [ 'internal', 'package', 'framework' ]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+      - uses: WillAbides/setup-go-faster@v1.8.0
+        with:
+          go-version: ${{ matrix.go }}
+      - uses: dominikh/staticcheck-action@v1.3.0
+        if: contains(fromJSON('["1.19", "1.20"]'), matrix.go)
+        with:
+          version:  '2023.1'
+          install-go: false
+          min-go-version: ${{ matrix.go }}
+          cache-key: ${{ matrix.go }}
+          working-directory: ${{ matrix.dir }}
+      - uses: dominikh/staticcheck-action@v1.3.0
+        if: contains(fromJSON('["1.17", "1.18"]'), matrix.go)
+        with:
+          version: '2022.1'
+          install-go: false
+          min-go-version: ${{ matrix.go }}
+          cache-key: ${{ matrix.go }}
+          working-directory: ${{ matrix.dir }}
```

Comment:
Just for my understanding, this was pulled out of the Makefile so we could run different staticcheck binaries (e.g. `2022.1` and `2023.1`) for different versions of Go?

Reply:
Yeah, it's done to integrate CI more GitHub-agnostic so we could leverage more of its feature like parallelization and visualization of jobs, and having a well defined set of supported Go-versions in that corresponding matrix for maintenance.

I also kept the target `go.staticcheck` in the Makefile for non-CI usage.

Technically, this would also be possible to achieve via changes in `Makefile` of course, but it still is more of an individual solution while GitHub-actions might be what other contributors are already used to work with so it lowers the initial barrier to get involved into contributing to `Bud`.